Apromore-Custom-Plugins/Log-Animation-Logic/src/main/java/org/apromore/service/loganimation/recording/Frame.java

```diff
@@ -121,12 +121,26 @@ public void removeToken(int logIndex, int tokenIndex) {
                 .toArray();
     }
     
-    public double getTokenDistance(int logIndex, int tokenIndex) {
+    /**
+     * This is the percentage from the start of the element (0..1) based on frame indexes
+     * This distance is suitable for the relative position of tokens on a modeling element
+     */
+    private double getFrameIndexRelativeTokenDistance(int logIndex, int tokenIndex) {
         int startFrameIndex = animationIndexes.get(logIndex).getStartFrameIndex(tokenIndex);
-        int numberOfFrames = animationIndexes.get(logIndex).getEndFrameIndex(tokenIndex) - startFrameIndex;
+        int endFrameIndex = animationIndexes.get(logIndex).getEndFrameIndex(tokenIndex);
+        int numberOfFrames = endFrameIndex - startFrameIndex;

```

Comment:
Is there an off-by-one error here?  If there is one frame at index 0 which is both the start frame and the end frame, then subtracting the two indices will give zero, not one.

Reply:
Ok. The variable name "numberOfFrames" is not totally accurate, changed it to maxLength. The purpose is when frameIndex = startFrameIndex, the return value is 0, when frameIndex = endFrameIndex, the return value is 1. The return value ranges from 0 to 1.